plugins/plugin-docker/che-plugin-openshift-client/src/main/java/org/eclipse/che/plugin/openshift/client/OpenShiftClientFactory.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2012-2017 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.plugin.openshift.client;
+
+import static io.fabric8.kubernetes.client.utils.Utils.isNotNullOrEmpty;
+
+import io.fabric8.kubernetes.client.Config;
+import io.fabric8.kubernetes.client.utils.HttpClientUtils;
+import io.fabric8.openshift.client.OpenShiftConfig;
+import io.fabric8.openshift.client.OpenShiftConfigBuilder;
+import io.fabric8.openshift.client.internal.OpenShiftOAuthInterceptor;
+import java.io.IOException;
+import javax.annotation.PreDestroy;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import okhttp3.Authenticator;
+import okhttp3.Interceptor;
+import okhttp3.Interceptor.Chain;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+
+@Singleton
+public class OpenShiftClientFactory {
+
+  private OkHttpClient httpClient;
+
+  @Inject
+  public OpenShiftClientFactory(
+      OpenshiftWorkspaceEnvironmentProvider workspaceEnvironmentProvider) {
+    this.httpClient =
+        HttpClientUtils.createHttpClient(workspaceEnvironmentProvider.getDefaultOpenshiftConfig());
+  }
+
+  @PreDestroy
+  public void cleanup() {
+    if (httpClient.connectionPool() != null) {
```

Comment:
As far as I see checking `httpClient.connectionPool()` on null is redundant at least for 3.6.0 http client https://github.com/square/okhttp/blob/parent-3.6.0/okhttp/src/main/java/okhttp3/OkHttpClient.java#L738. The same for `httpClient.dispatcher()` and `httpClient.dispatcher().executorService()`
Of course up to you to leave these checks or not.